File: speech-practice/android/app/src/main/java/app/sayitonce/BootReceiver.java
package app.sayitonce;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;

/** Alarms are cleared on reboot, so the reminder is scheduled again here. */
public class BootReceiver extends BroadcastReceiver {
    @Override
    public void onReceive(Context context, Intent intent) {
        if (Intent.ACTION_BOOT_COMPLETED.equals(intent.getAction())) {
            Reminders.schedule(context);
        }
    }
}

File: adhd-planner/app/src/main/java/com/groundcontrol/planner/alarm/BootReceiver.kt
package com.groundcontrol.planner.alarm

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import com.groundcontrol.planner.container
import kotlinx.coroutines.CoroutineScope
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.launch

/**
 * Alarms do not survive a reboot, a reinstall, or a clock change on their own.
 * Every one of those events re-lays the whole set.
 */
class BootReceiver : BroadcastReceiver() {

    override fun onReceive(context: Context, intent: Intent) {
        val pending = goAsync()
        val app = context.applicationContext
        CoroutineScope(Dispatchers.IO).launch {
            try {
                AlarmScheduler(app).syncAll(app.container.repository.all())
            } finally {
                pending.finish()
            }
        }
    }
}
